update_database_scripts/unpackDatabaseCommandLine.R

```diff
@@ -1,459 +1,488 @@
-                # remove rows missing risk alleles or odds ratios, or which have X or Y as their chromosome, or SNPs conditioned on other SNPs
-                studyData <- filter(studyData, !is.na(risk_allele)&!is.na(or_per_copy_number)&(or_per_copy_number > 0)&startsWith(variant_id, "rs")&!startsWith(hg38, "X")&!startsWith(hg38, "Y")&!grepl("condition", pvalue_description)&!grepl("adjusted for rs", pvalue_description))
-                # if there are not enough snps left in the study table, add it to a list of ignored studies
-                if (nrow(studyData) < minNumStudyAssociations) {
-                  invalidStudies <- c(invalidStudies, studyID)
-                  DevPrint(paste0("    Not enough studyData info for ", citation))
-                } else { # otherwise add the rows to the association table
-                  associationsTable <- bind_rows(studyData, associationsTable)
-                  studyIndeciesAppended <- c(studyIndeciesAppended, i)
-                }
-              }
-            }
-          }
-        }
-        # for every 10 studies, append to the associations_table.tsv
-        if (i %% 10 == 0) {
-          # if there are studies in the associations table, print them out and reset the tibble
-          if (nrow(associationsTable) > 0) {
-            associationsTable <- formatAssociationsTable(associationsTable)
-            appendToAssociationsTable(associationsTable)
-            # reset the associationsTable and keep going
-            associationsTable <- tibble()
-            indecesAppendedStr <- paste(studyIndeciesAppended,collapse=",")
-            DevPrint(paste0("Appended studies to output file: ", indecesAppendedStr, " of ", stopIndex))
-            studyIndeciesAppended <- c()
-          }
-          DevPrint(paste0("Time elapsed: ", format(Sys.time() - start_time)))
-        }
-      }
-    }, error=function(e){
-      cat("ERROR:",conditionMessage(e), "\n")
-      if (conditionMessage(e) == "cannot open the connection") {
-        stop("The table was likely opened during the download process and can't be used by the program. Please close the table and try again.")
-      }
-    })
-  }
-  
-  # if there are any studies left in the associationsTable, append them to the output file
-  if (nrow(associationsTable) > 0) {
-    associationsTable <- formatAssociationsTable(associationsTable)
-    appendToAssociationsTable(associationsTable)
-    indecesAppendedStr <- paste(studyIndeciesAppended,collapse=",")
-    DevPrint(paste0("Appended studies to output file: ", indecesAppendedStr, " of ", stopIndex))
-  }
-} else {
-  is_ebi_reachable(chatty = TRUE)
-  stop("The EBI API is unreachable. Check internet connection and try again.", call.=FALSE)
-}
-
-DevPrint(paste("Traits with no valid snps:", length(invalidStudies)))
-DevPrint(invalidStudies)
-print(paste0("Data download and unpack complete! ", "took: ", format(Sys.time() - start_time)))
\ No newline at end of file
+#!/usr/bin/env Rscript

```

Comment:
What lines are different in this file?

Reply:
This was one of the two most changed files. I fixed it so you can see the changes now.